### PR TITLE
Move the build of AOTriton to base ROCM docker image.

### PR DESCRIPTION
**Pull request by xinyazhang**
Mitigates #126111

AOTrtion, as a Math library, takes long time to build. However, this library itself is not moving as fast as PyTorch itself and it is not cost-efficient to build it for every CI check.

This PR moves the build of AOTriton from PyTorch to its base docker image, avoids duplicated and long build time.

Pre-this-PR:
* PyTorch base docker build job duration: 1.1-1.3h
* PyTorch build job duration: 1.4-1.5hr (includes AOTriton build time of 1hr6min on a linux.2xlarge node)

Post-this-PR:
* PyTorch base docker build job duration: 1.3h (includes AOTriton build time of 20min on a linux.12xlarge node)
* PyTorch build job duration: <20 min

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
Observations from CI:
AOTrtion took 50 mins to build in base docker image CI job: https://github.com/pytorch/pytorch/actions/runs/9229480020/job/25395796276

**Comment by jithunnair-amd** [comment]
Observations from latest CI runs:
[PyTorch CI base docker build](https://github.com/pytorch/pytorch/actions/runs/9283210346/job/25563757847): Took 1.3hrs on a `linux.12xlarge` node. Built AOTriton in <20mins
[PyTorch build job](https://github.com/pytorch/pytorch/actions/runs/9283211246/job/25542943887): Took <20 min (excluding CI base docker rebuild, which took 3hrs on a `linux.2xlarge` node)

File: .ci/docker/common/install_aotriton.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+git submodule sync --recursive
+git submodule update --init --recursive --force --depth 1
+mkdir build
+cd build
+cmake .. -G Ninja -DCMAKE_INSTALL_PREFIX=./install_dir -DCMAKE_BUILD_TYPE=Release -DAOTRITON_COMPRESS_KERNEL=OFF -DAOTRITON_NO_PYTHON=ON -DAOTRITON_NO_SHARED=ON
+ninja install
+mkdir -p "${AOTRITON_INSTALL_PREFIX}"
+cp -r install_dir/* "${AOTRITON_INSTALL_PREFIX}"
+find /tmp/ -mindepth 1 -delete
+rm -rf ~/.triton
```
Comment (jithunnair-amd) [inline .ci/docker/common/install_aotriton.sh:24]: This directory takes up 3.8G, and contains build intermediates and MLIR compiler.

**Review by jithunnair-amd — APPROVED** [review]

**Review by pruthvistony — APPROVED** [review]

**Comment by pruthvistony** [comment]
@malfet @huydhn @atalman ,
Can you please help review this PR.

**Comment by xinyazhang** [comment]
The two docker build failures (https://hud.pytorch.org/pytorch/pytorch/pull/127012?sha=fffac536c65e7cf8380d5ca25c54f06eae41e612) seems to be networking errors

**Review by huydhn — APPROVED** [review]
LGTM!

**Pull request closed without merging**